_overviews/scala3-contribution/arch-symbols.md

```diff
@@ -0,0 +1,133 @@
+ - reference: string + sig
+
+
+`dotc` is different from most other compilers in that it is centered around the idea of
+maintaining views of various artifacts associated with code. These views are indexed
+by time.
+
+A symbol refers to a definition in a source program. Traditionally,
+compilers store context-dependent data in a _symbol table_. The
+symbol then is the central reference to address context-dependent
+data. But for the requirements of `dotc` it turns out that symbols are
+both too little and too much for this task.
+
+**Too little:** The attributes of a symbol depend on the phase. Examples:
+Types are gradually simplified by several phases. Owners are changed
+in phases `LambdaLift` (when methods are lifted out to an enclosing
+class) and Flatten (when all classes are moved to top level). Names
+are changed when private members need to be accessed from outside
+their class (for instance from a nested class or a class implementing
+a trait). So a functional compiler, a `Symbol` by itself met mean
+much. Instead we are more interested in the attributes of a symbol at
+a given phase.
+
+**Too much:** If a symbol is used to refer to a definition in another
+compilation unit, we get problems for incremental recompilation. The
+unit containing the symbol might be changed and recompiled, which
+might mean that the definition referred to by the symbol is deleted or
+changed. This leads to the problem of stale symbols that refer to
+definitions that no longer exist in this form. Scala 2 compiler tried to
+address this problem by _rebinding_ symbols appearing in certain cross
+module references, but it turned out to be too difficult to do this
+reliably for all kinds of references. Scala 3 compiler attacks the problem at
+the root instead. The fundamental problem is that symbols are too
+specific to serve as a cross-module reference in a system with
+incremental compilation. They refer to a particular definition, but
+that definition may not persist unchanged after an edit.
+
+`dotc` uses instead a different approach: A cross module reference is
+always type, either a `TermRef` or `TypeRef`. A reference type contains
+a prefix type and a name. The definition the type refers to is established
+dynamically based on these fields.
+
+
+<!-- a system where sources can be recompiled at any instance,
+
+ the concept of a `Denotation`.
+
+ Since definitions are transformed by phases, -->
+
+## Symbols
+`dotc/core/Symbols.scala`
+
+Symbols are references to definitions (e.g. of variables, fields, classes). Symbols can be used to refer to definitions for which we don't have ASTs (for example, from the Java standard library).
+
+`NoSymbol` is used to indicate the lack of a symbol.
+
+Symbols uniquely identify definitions, but they don't say what the definitions *mean*. To understand the meaning of a symbol
+we need to look at its *denotation* (spefically for symbols, a `SymDenotation`).
```

Comment:
```suggestion
we need to look at its *denotation* (specially for symbols, a `SymDenotation`).
```

Reply:
@sjrd @vincenzobaz could you apply this change while reviewing? @bishabosha won't do it since he's on vacation and I don't have access to @scalacenter branches.

Reply:
Shouldn't it be "specifically" instead?

Reply:
"for symbols specifically" would be the clearest and most idiomatic, IMO